# A reversible state of hypometabolism in a cellular model of sporadic Parkinson’s disease

## Abstract
Sporadic Parkinson’s Disease (sPD) is a progressive neurodegenerative disorder caused by multiple genetic and environmental factors with largely unknown etiology. Prominent pathological culprits in patients are mitochondrial alterations. However, their relevance at different stages of disease progression remains largely elusive. Here, human induced pluripotent stem cells (hiPSCs) from late-onset sPD patients and healthy individuals were used for disease modeling. Following hiPSC differentiation, neural precursor cells and dopaminergic neurons derived from sPD patients exhibit mitochondrial complex I deficiency next to a state of hypometabolism. A multilayered omics analysis based on single-cell transcriptomics, proteomics, and metabolomics allowed us to identify the citric acid cycle on the level of the α-ketoglutarate dehydrogenase complex as being the bottleneck in sPD metabolism. In sum, this model integrates sPD-associated mitochondrial dysfunction with metabolic alterations into one comprehensive mechanistic model. Notably, a follow-up of disease state in patients about 10 years after initial biopsy reveals a high correlation between a-ketoglutarate dehydrogenase complex activity changes and disease progression. In addition, the alterations in cellular metabolism such as the glucose metabolization and the activity of the α-ketoglutarate dehydrogenase complex were restored by interfering with the enhanced SHH signal transduction mediated by sPD-specific alterations in primary cilia function. Thus, inhibiting overactive SHH signaling may contribute to a neuroprotective therapy during early stages of sPD.

## Requirements
Here we collect the code used to analyse our data. 

For R code, following packages were used within R 4.2.2:
<ul>
<li> car v3.0-10 </li> 
<li> clusterProfiler v4.2.2 </li> 
<li> corrplot v0.92 </li> 
<li> DESeq2 v 1.36.0 </li> 
<li> EnhancedVolcano version 1.12.0 </li> 
<li> factoextra v1.0.7 </li> 
<li> FactorMineR v2.8 </li> 
<li> fdrtool v1.2.15 </li> 
<li> ggbiplot v0.55 </li> 
<li> ggplot2 v3.4.0 </li> 
<li> ggpubr v0.6.0 </li> 
<li> gplots v3.1.1 </li> 
<li> Hmisc v5.1-0 </li> 
<li> lme4 v1.1-31 </li> 
<li> pheatmap v1.0.12 </li> 
<li> RCy3 v2.14.2 </li> 
<li> ReactomePA v1.38.0 </li> 
<li> sm v2.2-5.681 </li> 
<li> VennDiagram v1.6.20 </li> 
</ul>

For the visualization of WikiPathways terms in cytoscape v3.9.1:
<ul>
<li> WikiPathways v3.3.10 </li> 
<li> CyTargetLinker v4.1.0 </li> 
<li> stringApp v1.7.0 </li> 
<li> enhancedGraphics v1.5.4 </li> 
</ul>

For Matlab code, following packages were used within MATLAB R2018a:
<ul>
<li> Statistics and Optimization Toolbox </li> 
<li> INCA v2.1 [https://mfa.vueinnovations.com/] </li> 
</ul>

## Citation
